docs/misc/components.md

```diff
@@ -0,0 +1,118 @@
+Text Components
+==================
+
+A `Component` is a holder for text which can be formatted and chained with other components via its subtype `MutableComponent`.
+A component can be created using one of the available static helpers:
+
+| Method Name    | Description                                                                                                             |
+|----------------|-------------------------------------------------------------------------------------------------------------------------|
+| `literal`      | it creates a component which simply wraps passed in text.                                                               |
+| `nullToEmpty`  | it's same as `#literal` except it creates empty component if null has been passed                                       |
+| `translatable` | it creates a component which is represented as localized text to user, read [internationalization] for more details.    |
+| `empty`        | it creates an empty component                                                                                           |
+| `keybind`      | it creates a component which is represented as name of current Keyboard key of passed keybind.                          |
+| `nbt`          | it creates a component for representing nbt data specified by `path` inside `dataSource`                                |
+| `score`        | it creates a component for representing `objective`'s score of entity specified by [entity selector][selectors] `name`. |
+| `selector`     | it creates a component for displaying list of names of entities selected by [entity selector][selectors] `pattern`.     |
+
+Component's text contents are represented by `ComponentContents`, which holds data and defines how to represent it as text.
+These helpers create `ComponentContents`, and wrap it in a `MutableComponent`.
```

Comment:
Redundant sentence, can be removed.

Reply:
I don't know which one you're referring to, is it `These helpers create ComponentContents, and wrap it in a MutableComponent.`?

Reply:
Yes.